### PR TITLE
feat: Implement wallet manager

### PR DESCRIPTION
**Pull request by duanyytop**

### CODE REVIEW

File: packages/ui/src/containers/ChangePasswod/index.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  const handleSubmit = () => {
+    getWalletIndex().then(res => {
+      if (isSuccessResponse(res) && res.result.current) {
+        const id = res.result.current
+        updateWalletName({ id, name }).then(res => {
+          if (isSuccessResponse(res)) {
+            history.push(Routes.Main)
+          } else {
+            throw new Error(res.message)
```
Comment (Keith-CY) [inline packages/ui/src/containers/ChangePasswod/index.tsx:29]: The error is not handled.

File: packages/ui/src/containers/ChangePasswod/index.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import React from 'react'
```
Comment (Keith-CY) [inline packages/ui/src/containers/ChangePasswod/index.tsx:1]: file name has typo

File: packages/ui/src/containers/ChangeWalletName/index.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  const handleSubmit = () => {
+    getWalletIndex().then(res => {
+      if (isSuccessResponse(res) && res.result.current) {
+        const id = res.result.current
+        updateWalletName({ id, name }).then(res => {
+          if (isSuccessResponse(res)) {
+            history.push(Routes.Main)
+          } else {
+            throw new Error(res.message)
```
Comment (Keith-CY) [inline packages/ui/src/containers/ChangeWalletName/index.tsx:29]: error is not handled.

**Comment by duanyytop** [comment]
Every comment question has been updated in https://github.com/Keith-CY/keypering/pull/8/commits/3bcc4981d57d812f35d3ef0e8ce7d463f8a7fb93

File: packages/app/src/wallet/service.ts [hunk 4]
```diff
@@ -70,29 +72,58 @@ export const updateWallet = ({ id, name }: { id: string, name: string }) => {
...
+  if (!current) {
+    throw new CurrentWalletNotSetException()
   }
-
-  const keystorePath = getKeystorePath(id)
-  const keystore = JSON.parse(fs.readFileSync(keystorePath, 'utf8'))
-
-  if (!checkPassword(keystore, password)) {
-    throw new IncorrectPasswordException()
+  const pwdWindow = new PasswordWindow("Password", 'Input Password')
```
[the remark this answers is not in the document]
Comment (duanyytop) [reply packages/app/src/wallet/service.ts:80]: We can refer to the title of Neuron and will update it later

File: packages/app/src/wallet/service.ts [hunk 5]
```diff
@@ -108,21 +107,23 @@ export const exportKeystore = async() => {
...
         name: 'keystore',
         extensions: ['json'] 
     }],
     defaultPath: 'keystore',
     title: 'Export',
     buttonLabel: 'Export'
-  })).filePath as string
+  }))
+  if (canceled || typeof filePath !== 'string') {
+    throw new SelectDirectoryRejected()
```
Comment (Keith-CY) [inline packages/app/src/wallet/service.ts:125]: Em... I think skip next steps when the user clicks `cancel` is enough, it's one of basic interaction and should not be treated as an exception.

Comment (duanyytop) [reply packages/app/src/wallet/service.ts:125]: So should I just handle the error that the type of `filePath` is not `string` ?

Comment (Keith-CY) [reply packages/app/src/wallet/service.ts:125]: > So should I just handle the error that the type of `filePath` is not `string` ?

It's ok to check the type, but I think
```js
if (canceled) {
  return false
}
```
is enough to tell UI that the user has finished his interaction.

Comment (duanyytop) [reply packages/app/src/wallet/service.ts]: Updated in https://github.com/Keith-CY/keypering/pull/8/commits/3c49b5d31a77ab2812323be38969d2f93fdb7bcd

**Pull request merged**